quinn/src/endpoint.rs

```diff
@@ -244,12 +246,9 @@ impl Endpoint {
         let reason = Bytes::copy_from_slice(reason);
         let mut endpoint = self.inner.state.lock().unwrap();
         endpoint.connections.close = Some((error_code, reason.clone()));
-        for sender in endpoint.connections.senders.values() {
-            // Ignoring errors from dropped connections
-            let _ = sender.send(ConnectionEvent::Close {
-                error_code,
-                reason: reason.clone(),
-            });
+        for conn in endpoint.connections.refs.values() {
```

Comment:
Any chance this is racy - so that connections that have been slightly after this don't get the close signal? I guess not, because it all goes through the endpoint mutex. And the path to open a connection hopefully checks `endpoints.connection.close`?

Reply:
Yep; `Accept::poll` locks the same mutex, and attempts to connect will go through `ConnectionSet::insert` which is inside the same mutex and immediately `close`s the connection that it adds if `self.close` is set.